pandas/core/internals/array_manager.py

```diff
@@ -842,9 +842,7 @@ def iset(
         else:
             assert isinstance(loc, np.ndarray)
             assert loc.dtype == "bool"
-            # error: Incompatible types in assignment (expression has type "ndarray",
-            # variable has type "range")
-            indices = np.nonzero(loc)[0]  # type: ignore[assignment]
+            indices = np.array([i for i in range(len(loc)) if loc[i] is True])
```

Comment:
I think you would need to change line 837 to  `indices : range | np.ndarray = range(`

Mypy uses early binding, the if block declares the variable for the first time with type `range`. When mypy parses the else block, it thinks the type should be range and not np.ndarray. I believe pyright doesn't do that (for each block it uses its own type and then uses the union after an if/else)

Reply:
thanks, I get it. np.nonzero works much more effectively regarding time than my code changes.